packages/commonwealth/client/scripts/views/components/subscription_button.tsx

```diff
@@ -1,39 +1,60 @@
+import { CommunityAlert } from '@hicommonwealth/schemas';
+import { notifySuccess } from 'controllers/app/notifications';
 import React, { useState } from 'react';
-
-import { NotificationCategories } from '@hicommonwealth/shared';
-import { isNotUndefined } from 'helpers/typeGuards';
-import { CWButton } from 'views/components/component_kit/new_designs/CWButton';
-
 import app from 'state';
+import { useCreateCommunityAlertMutation } from 'state/api/trpc/subscription/useCreateCommunityAlertMutation';
+import { useDeleteCommunityAlertMutation } from 'state/api/trpc/subscription/useDeleteCommunityAlertMutation';
+import { CWButton } from 'views/components/component_kit/new_designs/CWButton';
+import { z } from 'zod';
+import { useUserStore } from '../../state/ui/user/user';
 
-export const SubscriptionButton = () => {
-  const subscriptions = app.user.notifications;
-  const communitySubscription = subscriptions.findNotificationSubscription({
-    categoryId: NotificationCategories.NewThread,
-    options: { communityId: app.activeChainId() },
-  });
+export const SubscriptionButton = ({
+  communityAlerts,
+}: {
+  communityAlerts: ReadonlyArray<z.infer<typeof CommunityAlert>> | undefined;
+}) => {
+  const user = useUserStore();
+  const existingAlert = communityAlerts?.some(
+    (a) => a.community_id === app.activeChainId(),
+  );
   const [notificationsOn, setNotificationsOn] = useState<boolean>(
-    isNotUndefined(communitySubscription),
+    existingAlert || false,
   );
 
+  const deleteCommunityAlert = useDeleteCommunityAlertMutation();
```

Comment:
please use this form 
`const { mutateAsync: deleteCommunityAlert } = useDeleteCommunityAlertMutation();`
here and in other usage of mutation hooks